src/core/server/csp/config.ts

```diff
@@ -7,28 +7,145 @@
  */
 
 import { TypeOf, schema } from '@kbn/config-schema';
+import { ServiceConfigDescriptor } from '../internal_types';
+
+interface DirectiveValidationOptions {
+  allowNone: boolean;
+  allowNonce: boolean;
+}
+
+const getDirectiveValidator = (options: DirectiveValidationOptions) => {
+  const validateValue = getDirectiveValueValidator(options);
+  return (values: string[]) => {
+    for (const value of values) {
+      const error = validateValue(value);
+      if (error) {
+        return error;
+      }
+    }
+  };
+};
+
+const getDirectiveValueValidator = ({ allowNone, allowNonce }: DirectiveValidationOptions) => {
+  return (value: string) => {
+    if (!allowNonce && value.startsWith('nonce-')) {
+      return `using "nonce-*" is considered insecure and is not allowed`;
+    }
+    if (!allowNone && (value === `none` || value === `'none'`)) {
+      return `using "none" would conflict with Kibana's default csp configuration and is not allowed`;
+    }
+  };
+};
+
+const configSchema = schema.object(
+  {
+    rules: schema.maybe(schema.arrayOf(schema.string())),
+    script_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
+    }),
+    worker_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
+    }),
+    style_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
+    }),
+    connect_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+    }),
```

Comment:
As discussed:
- we're forbidding `nonce-` for all directives
- we're forbidding `none` for the 3 directives with default values (`script-src`, `worker-src`, `style-src`)